src/uct/api/v2/uct_v2.h

```diff
@@ -293,26 +316,55 @@ typedef struct uct_md_mem_dereg_params {
      * in this mask will be ignored. Provides ABI compatibility with respect
      * to adding new fields.
      */
-    uint64_t                     field_mask;
+    uint64_t         field_mask;
 
     /**
      * Operation specific flags, using bits from @ref uct_md_mem_dereg_flags_t.
      */
-    unsigned                     flags;
+    unsigned         flags;
 
     /**
      * Memory handle to deregister.
      */
-    uct_mem_h                    memh;
+    uct_mem_h        memh;
 
     /**
      * Pointer to UCT completion object that is invoked when region is
      * invalidated.
      */
-    uct_completion_t             *comp;
+    uct_completion_t *comp;
 } uct_md_mem_dereg_params_t;
 
 
+typedef struct uct_md_mkey_pack_params {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_md_mkey_pack_field_mask_t. Fields not specified in this mask
+     * will be ignored. Provides ABI compatibility with respect to adding new
+     * fields.
+     */
+    uint64_t  field_mask;
+
+    /**
+     * Pack a remote key for this memory handle.
+     */
+    uct_mem_h memh;
```

Comment:
Based on the code the `memh` and `rkey_buffer` are not optional parameters and there for belong to the argument list and not the `uct_md_mkey_pack_params`

Reply:
In theory, `memh` can be replaced with something more light-weight in future for optimization purposes, as well  as `rkey_buffer` can be replaced with "in-place buffer", for example.. So, since there are options I would suggest to have it more flexible.

Reply:
`memh` is an opaque object and `key_buffer` is void *. Both provide quite a lot of flexibility in terms of what can reside behind the objects. Since we redesign the interface I suggest to bring all the proposal now.

Reply:
ok, fixed according to your suggestion

Reply:
@evgeny-leksikov thanks. I will do another pass over the implementation patch and I will try to improve documentation.